R-package/R/xgb.cv.R

```diff
@@ -134,17 +134,29 @@ xgb.cv <- function(params = list(), data, nrounds, nfold, label = NULL, missing
 
   check.custom.obj()
   check.custom.eval()
-
-  # Check the labels
-  if ((inherits(data, 'xgb.DMatrix') && is.null(getinfo(data, 'label'))) ||
-      (!inherits(data, 'xgb.DMatrix') && is.null(label))) {
-    stop("Labels must be provided for CV either through xgb.DMatrix, or through 'label=' when 'data' is matrix")
-  } else if (inherits(data, 'xgb.DMatrix')) {
-    if (!is.null(label))
-      warning("xgb.cv: label will be ignored, since data is of type xgb.DMatrix")
-    cv_label <- getinfo(data, 'label')
+  
+  # AFT uses 'label_lower/upper_bound' instead of 'label'
+  if (is.character(params$objective) && params$objective == 'survival:aft') {
+    if (!inherits(data, 'xgb.DMatrix')) {
+      stop("Objective 'survival:aft' requires the data to be an 'xgb.DMatrix'.")
+    }
+    if (is.null(getinfo(data, name = 'label_lower_bound')) || 
```

Comment:
Something I am wondering: most of these `getinfo` used throughout the function are only checking whether the DMatrix contains that field or not, but each call involves creating a full data copy just to check that it exists. @trivialfis Is there perhaps some function that could be used to just check if the DMatrix has the field or not? Do functions like `XGDMatrixGetFloatInfo` actually create a data copy when calling them?